src/sync.cpp

```diff
@@ -37,23 +39,26 @@ void PrintLockContention(const char* pszName, const char* pszFile, int nLine)
 //
 
 struct CLockLocation {
-    CLockLocation(const char* pszName, const char* pszFile, int nLine, bool fTryIn)
-    {
-        mutexName = pszName;
-        sourceFile = pszFile;
-        sourceLine = nLine;
-        fTry = fTryIn;
-    }
+    CLockLocation(
+        const char* pszName, const char* pszFile, int nLine, bool fTryIn, const std::string& thread_name) :
```

Comment:
style-nit (since you'd have to fix up other things anyway): Could move the colon to the next line, according to the style-guideline for new code?

```suggestion
        const char* pszName, const char* pszFile, int nLine, bool fTryIn, const std::string& thread_name)
            : fTry...
```